### PR TITLE
Feature: public profile screen

### LINKED ISSUE #96: Profile: create public profile screen
Create a public profile showing screen than can be used both for seeing one's profile (with an option to switch to edit mode), and for seeing the profile of other users (with a follow/unfollow button this time).

Integrate follower/following counts, make it clickable to display follower/following names.

### PR DESCRIPTION
**Pull request by robertaleG**
# Description

I have created a read-only public profile screen that was modularised and is used both for the current user's home profile screen (`UserProfileScreen`) and the public profile screen of any other user (`PublicProfileScreen`).  

Past changes to `ProfileScreen` were moved to the renamed file `EditProfileScreen` which contains the textfields and profile picture editor. As such, the `TopLevelDestinations.Profile` destination now leads to the user profile with a floating action button to go to the edit profile page.

~~NB: There is a placeholder image, that awaits PR #95 to implement real profile picture.~~
~~Edit: PR#95 is now merged, implementation in progress!~~
**Edit: profile pic implemented!**

Fixes #96 

### Other features added for QoL improvements:

- Biography for users (bio string field in User class)
- user handle, user full name and date to string member functions of User class
- delete button for profile pictures, redesign of delete and edit buttons

# How Has This Been Tested?

I have added ProfileScreenTest which tests all the mechanics of the UI: dropdown menus, links to other profile pages, follow/unfollow buttons, etc...

### Unit Tests
Expanded UserTest to test new member functions (handle, full name, dateToString)
  
### UI Tests
Added ProfileScreenTest.kt and fixed minor mistake in EditNoteTest.kt

### End to end Tests
No new End to End tests but now that the profile section is becoming large. An end to end test might be warranted. 

  
# Dependencies

None added

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged in downstream modules, no conflicts with main

### CODE REVIEW

File: app/src/main/java/com/github/onlynotesswent/ui/user/EditProfile.kt [hunk 1]
```diff
@@ -228,27 +217,47 @@ fun ProfilePicture(
...
+              imageVector = Icons.Default.Edit,
+              contentDescription = "Edit Profile Picture",
+              modifier = Modifier.testTag("editProfilePicture").size(40.dp) // Size of the edit icon
+              ,
+              tint = Color.Blue // Icon color
+              )
+        }
+
+    // Delete Image Icon
+    IconButton(
```
Comment (LawrenceLeitgib) [inline app/src/main/java/com/github/onlynotesswent/ui/user/EditProfile.kt:245]: Not sure if it's necessary to be able to delet a profile picture

File: app/src/main/java/com/github/onlynotesswent/ui/user/PublicProfile.kt [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+              Row {
+                Icon(Icons.Default.Create, contentDescription = "Edit Profile")
+                Spacer(modifier = Modifier.width(8.dp))
+                Text("Edit Profile")
+              }
+            }
+      }) {
+        ProfileContent(user, navigationActions, userViewModel, fileViewModel)
+
+        // Debug buttons to follow/unfollow specific users
```
Comment (LawrenceLeitgib) [inline app/src/main/java/com/github/onlynotesswent/ui/user/PublicProfile.kt:106]: we should delet those testing buttons before merging

**Review by LawrenceLeitgib — CHANGES_REQUESTED** [review]
Overall really good code, It's very well modularized, I request changes because of some little details below. Also the UI of edit profile is very ugly (the old one looked better) it's fine for now but we will need to improve it later

File: app/src/main/java/com/github/onlynotesswent/ui/user/EditProfile.kt (hunk 1 above)
Comment (robertaleG) [reply app/src/main/java/com/github/onlynotesswent/ui/user/EditProfile.kt:245]: yes it is necessary

File: app/src/androidTest/java/com/github/onlynotesswent/ui/user/EditProfileScreenTest.kt
[the remark this answers is not in the document]
Comment (robertaleG) [reply app/src/androidTest/java/com/github/onlynotesswent/ui/user/EditProfileScreenTest.kt:1]: we will modify it in a future pr, including a revised bottom sheet and profileHistory dependency implementing stacks

**Review by LawrenceLeitgib — APPROVED** [review]
Perfect

**Pull request merged**